src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -642,12 +647,29 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                         // If `falseReg` is zero, then move the first operand of `intrinEmbMask` in the
                         // destination using /Z.
 
-                        assert(targetReg != embMaskOp2Reg);
-                        GetEmitter()->emitIns_R_R_R(INS_sve_movprfx, emitSize, targetReg, maskReg, embMaskOp1Reg, opt);
+                        switch (intrinEmbMask.id)
+                        {
+                            case NI_Sve_CreateBreakPropagateMask:
+                                assert(targetReg != maskReg);
```

Comment:
nit: I don't believe this assert is "quite right".

`delayFree` is really just saying that in the case `targetReg != op2Reg` that no other operand can be the same as `targetReg` because we're going to need to overwrite `targetReg`.

However, in the case that `targetReg == op2Reg` then there is no reason that `targetReg == maskReg` can't also be true since no overwriting will happen.

Reply:
I see, so it's probably fine that I can remove the assert. I had it in their when I hit the case where the targetReg and maskReg were the same and causing issues.